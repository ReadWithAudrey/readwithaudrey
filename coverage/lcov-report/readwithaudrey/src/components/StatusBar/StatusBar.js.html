<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for readwithaudrey/src/components/StatusBar/StatusBar.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">readwithaudrey/src/components/StatusBar</a> StatusBar.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { Status, StatusActive } from './StatusBar.style'
import PropTypes from 'prop-types'
&nbsp;
const StatusBar = props =&gt; {
  if (props.type === 'active') {
    return &lt;StatusActive&gt;{props.children}&lt;/StatusActive&gt;
  } else {
    return &lt;Status&gt;{props.children}&lt;/Status&gt;
  }
}
&nbsp;
StatusBar.propTypes = {
  active: PropTypes.string,
  children: PropTypes.string,
}
&nbsp;
export default StatusBar
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 12 2018 12:51:26 GMT+0100 (British Summer Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
