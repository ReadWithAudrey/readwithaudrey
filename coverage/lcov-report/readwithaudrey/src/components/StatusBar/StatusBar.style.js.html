<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for readwithaudrey/src/components/StatusBar/StatusBar.style.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">readwithaudrey/src/components/StatusBar</a> StatusBar.style.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from 'styled-components'
&nbsp;
export const Status = styled.section.attrs({
  className: 'flex items-center fl w-third v-mid montserrat bg-moon-gray fw4 f5 mb3',
})`
  min-height: 8vh;
  color: #fff;
  padding: 1.5%;
  a {
    color: #fff;
    text-decoration: none;
  }
`
export const StatusActive = styled(Status).attrs({
  className: 'fl w-third v-mid bg-green montserrat',
})`
  color: #fff;
  font-weight: bold;
`
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 12 2018 12:51:26 GMT+0100 (British Summer Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
