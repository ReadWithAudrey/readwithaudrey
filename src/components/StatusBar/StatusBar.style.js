import styled from 'styled-components'

export const Status = styled.section.attrs({
  className: 'fl w-third v-mid montserrat bg-moon-gray fw4 f6',
})`
  height: 5vh;
  color: #fff;
  padding: 0%;
`
export const StatusActive = styled(Status).attrs({
  className: 'fl w-third  v-mid bg-green montserrat',
})`
  color: #fff;
`
