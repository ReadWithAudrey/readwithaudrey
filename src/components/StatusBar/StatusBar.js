import React from 'react'
import { Status, StatusActive } from './StatusBar.style'
import PropTypes from 'prop-types'

const StatusBar = props => {
  if (props.type === 'active') {
    return <StatusActive>{props.children}</StatusActive>
  } else {
    return <Status>{props.children}</Status>
  }
}

StatusBar.propTypes = {
  active: PropTypes.string,
  children: PropTypes.string,
}

export default StatusBar
