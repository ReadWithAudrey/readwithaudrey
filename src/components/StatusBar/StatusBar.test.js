import React from 'react'
import renderer from 'react-test-renderer'
import StatusBar from './StatusBar'

describe('StatusBar component snapshot test', () =>
  it('renders correctly', () => {
    const tree = renderer.create(<StatusBar />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
