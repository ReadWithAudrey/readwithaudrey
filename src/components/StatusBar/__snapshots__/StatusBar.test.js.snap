// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar component snapshot test renders correctly 1`] = `
<section
  className="sc-bdVaJa fl w-third v-mid montserrat bg-moon-gray fw4 f6 mb3 bpnKgt"
/>
`;
